### ISSUE TITLE
AMD example

### ISSUE THREAD

**Issue by mk-pmb** [systemjs/systemjs-examples#9]
Hi! Could someone please make an AMD example for me? I just can't get it to work. :-(

**Comment by internettrans** [comment]
Sure, I can submit a PR for this

**Comment by internettrans** [comment]
Here are two examples:

1. AMD Modules ([code](https://github.com/systemjs/systemjs-examples/tree/master/loading-code/amd-modules) and [live demo](https://systemjs.github.io/systemjs-examples/loading-code/amd-modules/))
2. Load Vue as an AMD dependency from CDN ([code](https://github.com/systemjs/systemjs-examples/tree/master/loading-dependencies/webpack-externals) and [live demo](https://systemjs.github.io/systemjs-examples/loading-dependencies/webpack-externals/))

I'll add another one real quick that shows usage of the named-exports extra, which can also be really important for AMD interop.

**Comment by internettrans** [comment]
Also see the AMD Dependencies example ([code](https://github.com/systemjs/systemjs-examples/tree/master/loading-dependencies/amd-dependencies) and [live demo](https://systemjs.github.io/systemjs-examples/loading-dependencies/amd-dependencies/)).

**Comment by mk-pmb** [comment]
Thanks a lot!

Edit: Your first example code link is broken, should probably be https://github.com/systemjs/systemjs-examples/tree/master/loading-code/amd-modules .

#### For other people who see just a blank page:
Looks like at least one of my problems was that [DecentralEyes](https://decentraleyes.org/) was configured to block requests for missing resources, and was missing the version of CDN files used in the example. So check in the network console whether your browser successfully receives plausible content for those requests.

**Comment by mk-pmb** [comment]
How about we add a file that explains the different role of the `src` and `lib` directories, add a hint comment to the top of all src files, and symlink it into each and `lib` directory as `how_to_build.md'?

**Comment by internettrans** [comment]
> Your first example code link is broken

I updated my comment

> How about we add a file that explains the different role of the src and lib directories, add a hint comment to the top of all src files, and symlink it into each and lib directory as `how_to_build.md'?

I don't think this is needed. Running `npm run build` is an industry standard way of building code with npm. Also, a `lib` directory consistently means "compiled code" and `src` means "uncompiled code" in the NodeJS community. We did not create those conventions and they have existed for several years. For those unfamiliar with those conventions, looking at the `scripts` inside of package.json and then running `npm run build` is a very reasonable thing to expect users to do.
